### PR TITLE
Add script repository in workbench

### LINKED ISSUE #25554: Implement ScriptRepository functionality in workbench
The script repository is quite frequently used, especially by Muons and Excitations groups. The functionality is required in the workbench also.

The existing GUI seems pretty stable but the code is not well tested. There were also questions at the user meeting as to whether further improvements such as file versioning, DOIs etc are worthwhile. On balance I think it would be better to wrap the existing GUI and expose it to Python and if future requirements need things this cannot support then we consider a rewrite then.

As part of this work it would be good to see if we can increase the unit testing of the code.

### PR DESCRIPTION
**Pull request by alicerussell1**
**Description of work.**
This PR adds the script repository into workbench. The functionality of it should be the same as in plot. A number of tests have also been added.

**To test:**
The [documentation ](https://www.mantidproject.org/ScriptRepository) contains the functionality of the script repo in plot. Test that it does the same in workbench.

Fixes #25554 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by Harrietbrown** [comment]
1. Having issues uploading, In some cases I am getting a server error, other cases it says the script has been uploaded successfully but it is not added to the git repository. Downloading works fine. We could to with some more people trying this to see who's end this is on. 

2. The description is not recognizing doc strings as it is supposed to based on the documentation.

3. The GUI window is very small by default, it could do with the top contents (the folder list) being taller by default.

**Comment by alicerussell1** [comment]
I have fixed the upload issue and that in turn should fix the doc strings as they only update when the file has been uploaded.

**Comment by Harrietbrown** [comment]
Great I'll test it out.

File: Framework/Kernel/CMakeLists.txt [hunk 1]
```diff
@@ -594,7 +594,7 @@ set(
   ${ExternalData_BINARY_ROOT}/Testing/Data/UnitTest;${ExternalData_BINARY_ROOT}/Testing/Data/DocTest;${MANTID_ROOT}/instrument
   )
 set(COLORMAPS_FOLDER ${MANTID_ROOT}/installers/colormaps/)
-set(MANTIDPUBLISHER "https://upload.mantidproject.org/scriptrepository?debug=1")
+set(MANTIDPUBLISHER "https://upload.mantidproject.org/scriptrepository")
```
Comment (martyngigg) [inline Framework/Kernel/CMakeLists.txt]: Can we leave this as `?debug=1` for developer builds. It prevents accidentally modifying the real repo and pushes the updates to https://github.com/mantidproject/sandbox.

Comment (alicerussell1) [reply Framework/Kernel/CMakeLists.txt]: Oh that makes sense, I hadn't realised that, it just looked like it wasn't uploading. I'll change it back

Comment (martyngigg) [reply Framework/Kernel/CMakeLists.txt]: For developer builds you can add a line to the `Mantid.user.properties` file to change it temporarily for production if required.

**Comment by Harrietbrown** [comment]
This is a really good addition to workbench, it works very well and is intuitive. Now the GUI is resized it is much easier to scroll through without resizing manualy.

Really great work.

**Review by Harrietbrown — APPROVED** [review]
This is a really good addition to workbench, it works very well and is intuitive. Now the GUI is resized it is much easier to scroll through without resizing manualy.

Really great work.

**Pull request merged**